internal/provider/resource_local_file.go

```diff
@@ -18,54 +18,64 @@ func resourceLocalFile() *schema.Resource {
 		Read:   resourceLocalFileRead,
 		Delete: resourceLocalFileDelete,
 
+		Description: "Generates a local file with the given content.",
+
 		Schema: map[string]*schema.Schema{
 			"content": {
-				Type:          schema.TypeString,
-				Optional:      true,
-				ForceNew:      true,
-				ConflictsWith: []string{"sensitive_content", "content_base64", "source"},
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ExactlyOneOf: []string{"content", "sensitive_content", "content_base64", "source"},
+				Description:  "Content to store in the file, expected to be an UTF-8 encoded string.",
 			},
 			"sensitive_content": {
-				Type:          schema.TypeString,
-				Optional:      true,
-				ForceNew:      true,
-				Sensitive:     true,
-				ConflictsWith: []string{"content", "content_base64", "source"},
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Sensitive:    true,
+				ExactlyOneOf: []string{"content", "sensitive_content", "content_base64", "source"},
+				Description:  "Sensitive content to store in the file, expected to be an UTF-8 encoded string.",
+				Deprecated:   "If in need to use _sensitive_ content, please use the `local_sensitive_file` resource instead.",
```

Comment:
Nit: Deprecation messages tend to just tell the practitioner what to do -- Terraform CLI should already provide the configuration file/line context pointing at the `sensitive_content` attribute in the warning diagnostic.

```suggestion
				Deprecated:   "Use the `local_sensitive_file` resource instead.",
```